framework/src/main/java/org/checkerframework/common/wholeprograminference/scenelib/AClassWrapper.java

```diff
@@ -0,0 +1,234 @@
+package org.checkerframework.common.wholeprograminference.scenelib;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.element.VariableElement;
+import javax.lang.model.type.TypeMirror;
+import org.checkerframework.checker.nullness.qual.MonotonicNonNull;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.checkerframework.common.wholeprograminference.SceneToStubWriter;
+import org.checkerframework.javacutil.BugInCF;
+import scenelib.annotations.Annotation;
+import scenelib.annotations.el.AClass;
+import scenelib.annotations.util.JVMNames;
+
+/**
+ * A wrapper for the AClass class from scene-lib that carries additional name and type information
+ * (referred to as "symbol information" elsewhere) that is useful during WPI.
+ *
+ * <p>This would be better as a subclass of AClass.
```

Comment:
Originally, because `AClass` is final and we didn't want to have to make changes to the AFU, because this PR should have just been able to use it as a library.

Mike and I have discussed this extensively, and eventually agreed not to change the code but to include this caveat, which is Mike's opinion on the matter. He has said that he might refactor this later so that it is, in fact, a subclass, as part of a larger project to improve the AFU.